backends/dpdk/dpdkArch.h

```diff
@@ -1101,10 +1101,23 @@ class ValidateOperandSize : public Inspector {
     }
 
     void postorder(const IR::Operation_Binary *binop) override {
+        if (binop->is<IR::BOr>() || binop->is<IR::BAnd>() || binop->is<IR::BXor>())
+            if (checkIf128bitOp(binop->left, binop->right)) return;
         isValidOperandSize(binop->left);
         isValidOperandSize(binop->right);
     }
 
+    bool checkIf128bitOp(const IR::Expression *src1Op, const IR::Expression *src2Op) {
+        if (auto src1Type = src1Op->type->to<IR::Type_Bits>()) {
+            if (src1Type->width_bits() == 128) {
+                if (auto src2Type = src2Op->type->to<IR::Type_Bits>()) {
+                    if (src2Type->width_bits() == 128) return true;
+                }
+            }
+        }
+        return false;
```

Comment:
Is it valid input to have operand1 of 128 bits and operand2 less than 128 bits at this stage?
If true, isValidOperandSize in line number 1106 may issue error for operand being greater than 64 bits.
IMO, isValidOperandSize function can be parameterized to check limit of 128 bits in exceptional cases.

Reply:
Typechecking will ensure that the operands to any binary operation like these are the exact same type, so there's really no need to check both.  The only binops that can have different types for their operands are shifts, array index, and concat.

Reply:
Updated the code.